src/backend/distributed/planner/recursive_planning.c

```diff
@@ -314,6 +314,11 @@ RecursivelyPlanSubqueriesAndCTEs(Query *query, RecursivePlanningContext *context
 		RecursivelyPlanAllSubqueries((Node *) query->jointree->quals, context);
 	}
 
+	if (query->havingQual != NULL)
+	{
+		RecursivelyPlanAllSubqueries((Node *) query->havingQual, context);
```

Comment:
Are there cases in which we can normally push this down? E.g. select on reference table?

Reply:
Might be best to fold this into `RecursivelyPlanNonColocatedSubqueries`. Will include a test that has a plan which differs between the two approachs

Reply:
Seems like we don't really support pushing down subqueries in having and the planner will end up doing the wrong thing if we allow selects on reference tables to go through here: The plan will select from the local reference table, rather than pushing down the subquery. Added a comment to #3433.